build-support/bin/make/mac/10.13/4.2.1/build-make.sh

```diff
@@ -0,0 +1,58 @@
+#!/bin/bash
+
+source "$(git rev-parse --show-toplevel)/utils.v1.bash"
+
+set_strict_mode
+
+function fetch_extract_make_source_release {
+  local -r extracted_dirname="make-${MAKE_VERSION}"
+  local -r archive_filename="${extracted_dirname}.tar.gz"
+  local -r release_url="https://ftpmirror.gnu.org/gnu/make/${archive_filename}"
+
+  local -r downloaded_archive="$(curl_file_with_fail "$release_url" "$archive_filename")"
+  extract_for "$downloaded_archive" "$extracted_dirname"
+}
+
+function build_make_with_configure {
+  local -a configure_cmd_line=("$@")
+
+  "${configure_cmd_line[@]}"
+
+  make "-j${SUBPROC_MAKE_JOBS}"
+
+  make install
+}
+
+function build_make_out_of_tree {
+  local -r source_extracted_abs="$(fetch_extract_make_source_release)"
+  local -r build_dir_abs="$(mkdirp_absolute_path 'make-build')"
+  local -r install_dir_abs="$(mkdirp_absolute_path 'make-install')"
+
+  with_pushd >&2 "$build_dir_abs" \
+             build_make_with_configure  \
+             "${source_extracted_abs}/configure" \
+             --prefix="$install_dir_abs"
+
+  with_pushd "$install_dir_abs" \
+             create_gz_package 'make'
+}
+
+## Interpret arguments and execute build.
+
+readonly MAKE_VERSION="$1"
+# Default to 2 parallel jobs if unspecified.
+readonly SUBPROC_MAKE_JOBS="${SUBPROC_MAKE_JOBS:-2}"
+
+case "$(uname)" in
```

Comment:
This script appears to be identical to `build-support/bin/make/linux/x86_64/4.2.1/build-make.sh` and yet still attempts to handle both linux and osx. Can there just be 1 script or else 2 different - smaller - scripts?

Reply:
There can be 1 or 2 such scripts (not sure of the ideal number yet), and there is no rush or other reason to justify not attempting to address this somewhat here.